### PR TITLE
Raise validation error on invalid start function index

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/unittests/parser_test.cpp [hunk 1]
```diff
@@ -781,15 +781,15 @@ TEST(parser, start_invalid_index)
                      make_section(3, func_section) + make_section(8, start_section) +
                      make_section(10, code_section);
 
-    EXPECT_THROW_MESSAGE(parse(bin), parser_error, "invalid start function index");
```
Comment (gumb0) [inline test/unittests/parser_test.cpp:784]: these tests should better be moved to `validation_test.cpp` now

Comment (axic) [reply test/unittests/parser_test.cpp:784]: Hm, I did that quite a few times. Perhaps we should merge these small changes as is and then do a sweep of moving things across?

**Comment by axic** [comment]
Also fixed stat function type checking -- should that be in a new PR or extend this one?

File: test/unittests/parser_test.cpp (hunk 1 above)
Comment (gumb0) [reply test/unittests/parser_test.cpp:797]: Fine

Comment (chfast) [reply test/unittests/parser_test.cpp:797]: Let's coordinate with moving bigger code pieces.

**Pull request merged**